### ISSUE TITLE
Support Serialize/deserialize functions in Type Adaption

### ISSUE THREAD

**Issue by audrow** [ros2/rclcpp#1665]
Currently, we only support convert functions to go between custom and ROS message types, which are more efficient for intra-process communication. It would be more efficient (less copy operations) for inter process communication to support serialization and deserialization functions.

<hr>

I've put this issue on the H-turtle board for consideration.

**Comment by wjwwood** [comment]
No, it would not. That's orthogonal to this issue which is supporting (de)serialization of custom types directly to/from the wire format. It doesn't cover other targets like yaml and it doesn't cover that for existing ROS types, only the custom types (which don't have `to_yaml()` either atm).
